go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -283,10 +283,15 @@ func runOvnKube(ctx *cli.Context) error {
 		metrics.StartMetricsServer(config.Kubernetes.MetricsBindAddress, config.Kubernetes.MetricsEnablePprof)
 	}
 
-	// start the prometheus server to serve OVN Metrics (default port: 9476)
+	// start the prometheus server to serve OVN Metrics
 	// Note: for ovnkube node mode dpu-host no ovn metrics is required as ovn is not running on the node.
 	if config.OvnKubeNode.Mode != types.NodeModeDPUHost && config.Kubernetes.OVNMetricsBindAddress != "" {
-		metrics.RegisterOvnMetrics(ovnClientset.KubeClient, node)
+		if master != "" {
```

Comment:
Right... I made this issue -> https://github.com/ovn-org/ovn-kubernetes/issues/2713 to track this 

Short of it is @girishmg doesn't run the ovn-dbs necessarily on the same nodes as the ovnkub-master like described here https://github.com/ovn-org/ovn-kubernetes/pull/2697#discussion_r769117735 so we can't only export metrics on the ovnkube-master nodes if I'm understanding correctly

Reply:
I see. This PR doesn't make sense so. Thanks Andrew.